packages/image_picker/image_picker_macos/macos/image_picker_macos/Sources/image_picker_macos/ImageCompress.swift

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import FlutterMacOS
+import Foundation
+
+/// Determines if the image should be compressed based on the quality.
+///
+/// - Parameter quality: The quality level (0-100). A quality less than 100 indicates compression.
+/// - Returns: Whether the image should be compressed.
+func shouldCompressImage(quality: Int64?) -> Bool {
+  return quality != nil && quality != 100
+}
+
+extension NSImage {
+  /// Compresses the image to the specified quality.
+  ///
+  /// - Parameter quality: The quality of the image (0 to 100).
+  /// - Returns: An optional `NSImage` that represents the compressed image.
+  func compressed(quality: Int64) throws -> NSImage {
+    assert(quality != 100, "Quality 100 means no compression.")
+    assert(quality >= 0, "Quality can't be negative.")
```

Comment:
Should I remove the asserts from the platform side since it will crash the app when running using `flutter run -d macos`?